sse2neon.h

```diff
@@ -8358,6 +8359,530 @@ FORCE_INLINE int _mm_testz_si128(__m128i a, __m128i b)
+                                        int la,
+                                        int lb,
+                                        __m128i mtx[16])
+{
+    int res = 0;
+    int m1 = 0x10000 - (1 << la);
+    uint8x8_t vec_mask = vld1_u8(_sse2neon_cmpestr_mask8b);
+    uint8x16_t vec1 = vcombine_u8(vtst_u8(vdup_n_u8(m1), vec_mask),
+                                  vtst_u8(vdup_n_u8(m1 >> 8), vec_mask));
+    uint8x16_t vec_minusone = vdupq_n_u8(-1);
+    uint8x16_t vec_zero = vdupq_n_u8(0);
+    for (int j = 0; j < lb; j++) {
+        mtx[j] = vreinterpretq_m128i_u8(
+            vbslq_u8(vec1, vec_minusone, vreinterpretq_u8_m128i(mtx[j])));
+    }
+    for (int j = lb; j < bound; j++) {
+        mtx[j] = vreinterpretq_m128i_u8(vbslq_u8(vec1, vec_minusone, vec_zero));
+    }
+    unsigned char *ptr = (unsigned char *) mtx;
+    for (int i = 0; i < bound; i++) {
+        int val = 1;
+        for (int j = 0, k = i; j < bound - i && k < bound; j++, k++) {
+            val &= ptr[k * bound + j];
+        }
+        res += val << i;
+    }
+    return res;
+}
+
+static int aggregate_equal_ordered_16x8(int bound,
+                                        int la,
+                                        int lb,
+                                        __m128i mtx[16])
+{
+    int res = 0;
+    int m1 = 0x100 - (1 << la);
+    uint16x8_t vec_mask = vld1q_u16(_sse2neon_cmpestr_mask16b);
+    uint16x8_t vec1 = vtstq_u16(vdupq_n_u16(m1), vec_mask);
+    uint16x8_t vec_minusone = vdupq_n_u16(-1);
+    uint16x8_t vec_zero = vdupq_n_u16(0);
+    for (int j = 0; j < lb; j++) {
+        mtx[j] = vreinterpretq_m128i_u16(
+            vbslq_u16(vec1, vec_minusone, vreinterpretq_u16_m128i(mtx[j])));
+    }
+    for (int j = lb; j < bound; j++) {
+        mtx[j] =
+            vreinterpretq_m128i_u16(vbslq_u16(vec1, vec_minusone, vec_zero));
+    }
+    unsigned short *ptr = (unsigned short *) mtx;
+    for (int i = 0; i < bound; i++) {
+        int val = 1;
+        for (int j = 0, k = i; j < bound - i && k < bound; j++, k++)
+            val &= ptr[k * bound + j];
+        res = (val << i) + res;
+    }
+    return res;
+}
+
+static int _sse2neon_cmp_byte_equal_ordered(__m128i a,
+                                            int la,
+                                            __m128i b,
+                                            int lb)
+{
+    __m128i mtx[16];
+    PCMPSTR_EQ_8x16(a, b, mtx);
+    return aggregate_equal_ordered_8x16(16, la, lb, mtx);
+}
+
+static int _sse2neon_cmp_word_equal_ordered(__m128i a,
+                                            int la,
+                                            __m128i b,
+                                            int lb)
+{
+    __m128i mtx[16];
+    PCMPSTR_EQ_16x8(a, b, mtx);
+    return aggregate_equal_ordered_16x8(8, la, lb, mtx);
+}
+
+typedef int (*cmpestr_func_t)(__m128i a, int la, __m128i b, int lb);
+
+static cmpestr_func_t _sse2neon_cmpfunc_table[] = {
```

Comment:
Embrace C99 array initialization. i.e.,
```c
static const cmpestr_func_t _sse2neon_cmpfunc_table[] = {
    [CMP_UBYTE_EQUAL_ANY] = _sse2neon_cmp_byte_equal_any,
    [CMP_UWORD_EQUAL_ANY] = _sse2neon_cmp_word_equal_any,
    ...
};

```

Reply:
By the way, [X macro](https://en.wikipedia.org/wiki/X_Macro) is a useful technique. Check https://github.com/sysprog21/rv32emu/blob/master/src/syscall.c and look for `__SYSCALL_LIST_BASE` for details.